libmodernize/main.py

```diff
@@ -28,6 +29,14 @@ def format_usage(usage):
     return usage
 
 def main(args=None):
+    old_os_linesep = os.linesep
+    try:
+        return actual_main(args)
+    finally:
+        if sys.version_info < (3, 0):
+            os.linesep = old_os_linesep
```

Comment:
If a test runs with `--windows-line-endings`, it will monkeypatch `refactor._to_system_newlines` to convert everything to Windows newlines on write. Then any subsequent test which doesn't specify its newlines will still write files with the Windows style newlines. For most tests, that won't matter, but for instance if your `test_mixed_to_native_line_endings` ran immediately after `test_unix_to_windows_line_endings`, I think it would fail.

I'm just finishing up a PR to achieve the same thing without monkeypatching, though.